crates/red_knot_workspace/resources/test/corpus/25_func_annotations_same_name.py

```diff
@@ -0,0 +1,11 @@
+def bool(x) -> bool:
+    return True
+
+
+class MyClass: ...
+
+
+def MyClass() -> MyClass: ...
+
+
+def x(self) -> x: ...
```

Comment:
In an ideal world, I'd prefer to have inference tests that validate we get the semantics right, rather than just corpus tests that validate we don't panic.

But at the moment it'd be a pain to write those inference tests; we don't have `Call` support yet, so you can't easily get access to the return type of a function by e.g. doing `y = x()` and then checking the type of the `y` symbol, you'd have to instead scrabble through the AST to find the return annotation expression node, and then assert on its type. So this is more just a note for future, not something that needs to be changed in this PR.